apps/04TowersOfHanoi.js

```diff
@@ -18,31 +18,44 @@ function printStacks() {
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
-
-function movePiece() {
+//Popping and pushing the startstack and endstack so the pieces will move.
+function movePiece(startStack, endStack) {
   // Your code here
+  var begin = stacks[startStack].pop();
+  stacks[endStack].push(begin);
 
 }
 
-function isLegal() {
-  // Your code here
-
+//Making sure the moves are legal moves.
+function isLegal(startStack, endStack) {
+  var start = stacks[startStack].slice(-1)
+  var end = stacks[endStack].slice(-1)
+  if (stacks[endStack].length === 0) {
+    return true
+  }
+  return start < end
```

Comment:
This is a nice clean function, but it might surprise you to learn that `slice` on line 31-2 is returning an array! So the comparison here is doing, e.g. `[1] < [2]` which is definitely not what we want (although it's a REALLY weird piece of JS behavior that actually works most of the time, hence your passing tests :D)